src/compiler/checker.ts

```diff
@@ -8069,7 +8069,8 @@ namespace ts {
             if (containsSameNamedThisProperty(expression.left, expression.right)) {
                 return anyType;
             }
-            const type = resolvedSymbol ? getTypeOfSymbol(resolvedSymbol) : getWidenedLiteralType(checkExpressionCached(expression.right));
+            const isDirectExport = kind === AssignmentDeclarationKind.ExportsProperty && (isPropertyAccessExpression(expression.left) || isElementAccessExpression(expression.left)) && (isModuleExportsAccessExpression(expression.left.expression) || (isIdentifier(expression.left.expression) && isExportsIdentifier(expression.left.expression)));

```

Comment:
```suggestion
            const isDirectExport = kind === AssignmentDeclarationKind.ExportsProperty 
                && isAccessExpression(expression.left)
                && (isModuleExportsAccessExpression(expression.left.expression) || (isIdentifier(expression.left.expression) && isExportsIdentifier(expression.left.expression)));
```

I think there might be an existing utility that does what you want here, but it may include other cases. (Could it be `isCommonJsExportPropertyAssignment`?)

Reply:
I couldn't find one. :S

Reply:
It would still be nice to (1) switch to `isAccessExpression` (2) break up this super-long long